### PR TITLE
Improves user search per #618

### PR DESCRIPTION
**Pull request by Razesdark**
Switches from search.list to channel.list api endpoint for user search.

### CODE REVIEW

**Comment by ids1024** [comment]
Yes, that's the way to do it, and it seems to work properly. I think it's probably best not to merge until searching is implemented (which I presume is your intent in tagging it WIP) since it breaks imprecise user searches.

**Comment by Razesdark** [comment]
Ok, I've been making some potential progress this week. I have spent some time making a content agnostic listview. The code is stilll pretty hacked together, and not very delicate so I won't upload until its a bit cleaner. But since I can't work on this for a week or so, I'll write it up so I can get some feedback on it.

Here is the current user search view. (Plus some debugstrings i forgot to clean up).
test

I've been trying to work on a content agnostic list view and a few supporting functions and classes, reusing a lot of the generate_playlist_code. It differs a little bit from how PaginatedContent was expected to work as this view remembers its own page number. I also made it so that a function named _run() can be added to any content.Content derived classes which gets priority when matching userinput, so future uses of content.Content can hold its own functions. I use this to change pages. When you create a view, you can specify which function it is going to run when you've selected your candidate from the list.

I also made a wrapper class for call_gdata, that emulates a list of all search results, but it will only query for more pages when needed.

**Comment by ids1024** [comment]
That generally sounds good, though I can't comment on the exact details without seeing the code.

The content API is a work in progress, and certainly not perfect, so any improvement is welcome. Originally `g.content` contained just strings. In #423 I tried to get the content API to be used for everything, but that stalled because it was somewhat complicated and I've been working on other projects.

**Comment by Razesdark** [comment]
Alright, I can take a look into that commit over the next few weeks and the code and see what I can do about it. I see some of those points will end up being covered by ListView, potentially.

**Comment by ids1024** [comment]
> So a single content class that we build subclasses for each kind of view from, making a unified way of rendering and interfacing with those derived classes so we have all code relevant to a single view in a single location would be a OK way to move forward?

That sounds good. Keep in mind that we want to move to a curses-based interface (which is why I began refactoring with a `Content` class a while ago), so it needs to be compatible with that. I think this should work for that (perhaps the `.content()` method could be done differently with a curses ui, but the important thing is that the relevant code is all in one place.)

File: mps_youtube/commands/search.py [hunk 1]
```diff
@@ -151,14 +150,40 @@ def channelfromname(user):
...
+    columns = [
+        {"name": "idx", "size": 3, "heading": "Num"},
+        {"name": "name", "size": 30, "heading": "Username"},
+        {"name": "description", "size": "remaining", "heading": "Description"},
+        ]
+
+    def run_m(user_id):
+        """ Search ! """
+        usersearch_id(*user_id)
+    del g.content
```
Comment (ids1024) [inline mps_youtube/commands/search.py]: Why would this be necessary?

File: mps_youtube/listview.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
+"""
+    DOCSTING COMES HERE
+"""
+import re
+
+from . import c, util, content
+
+
+class ListViewItem(object):
```
Comment (ids1024) [inline mps_youtube/listview.py]: Mpsyt is now Python 3 only, so explicitly inheriting from `object` is redundant.

File: mps_youtube/commands/search.py (hunk 1 above)
Comment (Razesdark) [inline mps_youtube/commands/search.py]: It isn't. I found a bug in the code where a second channel search would still yield the results from the first search. I wasn't entirely sure what caused this, so I added some code to make sure i deleted previous data before rewriting them, . When I found the bug, i removed much of that code, seems I skipped this line.

File: mps_youtube/listview.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        types = len(set([obj.__class__ for obj in objects]))
+        if types == 0:
+            raise BaseException("No objects in list")
+        if types > 1:
+            raise BaseException("More than one kind of objects in list")
+
+        self.object_type = [obj.__class__ for obj in objects][0]
+        del types
+
+    def pages(self):
```
Comment (ids1024) [inline mps_youtube/listview.py]: This function is just the equivalent of `max(1, len(self.objects) / self.views_per_page())`, right?

Comment (ids1024) [inline mps_youtube/listview.py]: Plus a call to `math.ceil`

Comment (Razesdark) [inline mps_youtube/listview.py]: Yea, ceil was was i was going for.

**Comment by ids1024** [comment]
Merged with some changes; thanks! I made ListView a subclass of `PaginatedContent` (8779160ae185680bc80cc35e0a7b02d19559c86c). I agree that `g.current_page` should be part of the content API rather than a global, but that can be dealt with when a content API based solution is used for playlist search and `paginatesongs()`, so there is no duplication.

**Pull request merged**